devicetypes/smartthings/ezex-smart-electric-switch.src/ezex-smart-electric-switch.groovy

```diff
@@ -0,0 +1,120 @@
+ *
+ */
+metadata {
+    definition (name: "eZEX smart electric switch", namespace: "smartthings", author: "SmartThings", mnmn: "SmartThings", vid: "generic-switch-power-energy") {
+        capability "Energy Meter"
+        capability "Power Meter"
+        capability "Actuator"
+        capability "Switch"
+        capability "Refresh"
+        capability "Health Check"
+        capability "Sensor"
+
+        fingerprint profileId: "0104", deviceId:"0053", inClusters: "0000, 0003, 0004, 0006, 0B04, 0702", outClusters: "0019", manufacturer: "", model: "E240-KR116Z-HA", deviceJoinName: "Smart Electric Switch"
+    }
+
+    tiles(scale: 2){
+        multiAttributeTile(name:"switch", type: "generic", width: 6, height: 4, canChangeIcon: true){
+                tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+                        attributeState("on", label: '${name}', action: "switch.off", icon: "st.switches.switch.on", backgroundColor: "#00a0dc")
+                        attributeState("off", label: '${name}', action: "switch.on", icon: "st.switches.switch.off", backgroundColor: "#ffffff")
+                }
+        }
+        valueTile("power", "device.power", decoration: "flat", width: 2, height: 2) {
+                state "default", label:'${currentValue} W'
+        }
+        valueTile("energy", "device.energy", decoration: "flat", width: 2, height: 2) {
+                state "default", label:'${currentValue} kWh'
+        }
+        standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+                state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+        standardTile("reset", "device.energy", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+                state "default", label:'reset kWh', action:"reset"
+        }
+
+        main(["switch"])
+        details(["switch","power","energy","refresh","reset"])
+    }
+}
+
+def parse(String description) {
+    log.debug "description is $description"
+    def event = zigbee.getEvent(description)
+    if (event) {
+        log.info event
+        sendEvent(event)
+    } else {
+        List result = []
+        def descMap = zigbee.parseDescriptionAsMap(description)
+        log.debug "Desc Map: $descMap"
+        
+        List attrData = [[clusterInt: descMap.clusterInt ,attrInt: descMap.attrInt, value: descMap.value]]
+        descMap.additionalAttrs.each {
+            attrData << [clusterInt: descMap.clusterInt, attrInt: it.attrInt, value: it.value]
+        }
+        attrData.each {
+                def map = [:]
+                if (it.clusterInt == 0x0702 && it.attrInt == 0x0400) {
+                        log.debug "power"
+                        map.name = "power"
+                        map.value = zigbee.convertHexToInt(it.value)/1000
+                }
+                if (it.clusterInt == 0x0702 && it.attrInt == 0x0000) {
```

Comment:
A couple of things around here.

This should be an `else if`

Instead of using `0x0702`, let's use `zigbee. SIMPLE_METERING_CLUSTER` and for the attrInt hex numbers, let's use constants like https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/8941ead2562344afa95a8818ec8462068f230b6d/devicetypes/stelpro/stelpro-ki-zigbee-thermostat.src/stelpro-ki-zigbee-thermostat.groovy#L241

(Something like `ATTRIBUTE_READING_INFO_SET` for `0x0000` and `ATTRIBUTE_HISTORICAL_CONSUMPTION` for `0x0400` and don't forget to declare them with `def getATTRIBUTE_READING_INFO_SET() { 0x0000 }` and `def getATTRIBUTE_HISTORICAL_CONSUMPTION() { 0x0400 }`)

This just makes it a little cleaner. (We should have caught that for the other DTH.)